File: frontend/gestionale-bertozzi/src/app/models/GestioneCommesse/piano-sviluppo.ts
import { Attivita } from './attivita';

export class PianoSviluppo {
    id?: number;
    commessaId?: number;
    descrizione?: string;
    ordine?: number;
    dataCreazione?: Date;
    dataModifica?: Date;
    utenteCreazione?: string;
    utenteModifica?: string;
    attivita?: Attivita[];

    static map(piano: any): PianoSviluppo {
        let p = Object.assign(new PianoSviluppo(), piano) as PianoSviluppo;
        
        // Mappa l'array delle attività se presente
        if (piano.attivita && Array.isArray(piano.attivita)) {
            p.attivita = Attivita.mapArray(piano.attivita);
        }
        
        return p;
    }

    static mapArray(arr: any[]): PianoSviluppo[] {
        return (arr || []).map(p => PianoSviluppo.map(p));
    }
}

File: backend/NemesiLIB/Model/GestioneCommesse/PianoSviluppo.cs
using NemesiCOMMONS.Models;
using System;
using System.Collections.Generic;

namespace NemesiLIB.Model.GestioneCommesse
{
    public class PianoSviluppo : IAuditable
    {
        public int Id { get; set; }
        public int CommessaId { get; set; }
        public string Descrizione { get; set; }
        public int Ordine { get; set; }
        public DateTime DataCreazione { get; set; }
        public DateTime? DataModifica { get; set; }
        public string? UtenteCreazione { get; set; }
        public string? UtenteModifica { get; set; }

        // Navigation properties
        public virtual List<Attivita>? Attivita { get; set; }
        public virtual List<OreSpeseCommessa>? OreSpese { get; set; }
    }
}